8.0 migrate from 7.0

lmignon opened the pull request:
Current status:
- [x] Forward port changes from PR #16 
- [x] Defines Fields for specialized geo columns 
- [x] Migrate to the new API
- [x] Extract description in __openerp__.py to README.rst
- [x] Use an init hook to initialize postgis extension at install
- [x] Move icons to static/description
- [x] Fix issues when google is used as Raster Layer Type (https://github.com/acsone/geospatial/commit/6f924a2d7df99a7c956851ddd499fdb6db22a4f4 should be backported in 7.0)
- [x] ~~Fix issues when the map is displayed for a specific field. (the label is not visible the map takes the whole width)~~ The view definition in the demo data was wrong. (see https://github.com/acsone/geospatial/commit/9d939da17211b8ff1e787b3c6475fbbae5e1bc01)

At this stage addons are usable in V8 but some remaining issues are to be fixed. However, two bugs in Odoo prevent proper use. To work, you need to merge the following 2 branches.
-   https://github.com/acsone/odoo/tree/8.0-fix-5616-ir_model_fields-from-fields (PR odoo/odoo/pull/5618)
-  https://github.com/acsone/odoo/tree/8.0-fix-use_symbol-c-in-sql-create (PR odoo/odoo/pull/5620)

nbessi committed:
Adding .gitignore file

nbessi committed:
Adding README.md file

nbessi committed:
Fix non protected import of python dependences

nbessi committed:
Fix import protection in geo helpers

yvaucher committed:
Merge pull request #3 from nbessi/fix_import

Fix non protected import of python dependences

lmignon committed:
Make modules intallable

lmignon committed:
[MIG] geospatial widgets

lmignon committed:
forward port change from 6.1

lmignon committed:
[FIX] geonames.org geocoder requires a valid username

lmignon committed:
[FIX] correctly support readonly attribute on fields.geo_point

lmignon committed:
Put only two numbers for version.

lmignon committed:
self.pool.get -> self.pool[

lmignon committed:
[IMP] Sample module works fine on 7.0

lmignon committed:
[ADD] travis and LICENSE

lmignon committed:
[PEP8] base_geoengine

lmignon committed:
[FIX] postgis extension initialization

lmignon committed:
[IMP] Add travis and coverage badges

lmignon committed:
[IMP] no UNIT_TEST

lmignon committed:
Calling travis_run_flake8 is deprecated now that it's done in travis_run_tests

lmignon committed:
[IMP] Add requirements

lmignon committed:
[IMP] naming file consistency

lmignon committed:
[PEP8] base_geoengine_demo

lmignon committed:
[PEP8] geoengine_geoname_geocoder

lmignon committed:
[PEP8] geoengine_partner

lmignon committed:
[PEP8] geoengine_sale

lmignon committed:
[PYLINT] base_geoengine

lmignon committed:
[IMP] Better formatting of requirements

lmignon committed:
[IMP] Add basic tests

lmignon committed:
Merge branch '7.0-pep8' of https://github.com/acsone/geospatial into 7.0-pep8

lmignon committed:
[FIX] forward port 7.0

lmignon committed:
[FIX] move all addons from __unported__

lmignon committed:
[IMP] Add licence and fix build version

lmignon committed:
[FIX] Uses the new API

lmignon committed:
[IMP] Use the non minified version of Openelayers.js
Js are minified by Odoo in normal mode but keep as initial in debug mode

lmignon committed:
[FIX] Use absolute imports on opnerp.addons

lmignon committed:
[FIX] Defines Fields for our specialized columns

lmignon committed:
[FIX] Follows V8 conventions
remove deprecated entries
include js dependencies in a override of web.assets template

lmignon committed:
[FIX] View inheritance

lmignon committed:
[FIX] Properly displays the map
* Renders the map only when its element is displayed since the size must be known to avoid errors
* Remove unused pager
* Fix declared style for svg elements. The size must be known and fixed for svg elements of map.

nbessi commented:
@lmignon  do not forget to extact the test function to a unit test

lmignon commented:
@nbessi it's already done see https://github.com/acsone/geospatial/commit/2ddc957008ddc71dd667ad702cdb4291e8a1092e

nbessi commented:
I just saw it @lmignon  too fast  :racehorse:

nbessi commented:
@lmignon We should also try to migrate to OpenLayers3 we should get a lot of benefit to use it. 

Maybe we can do it in an other PR.

lmignon commented:
@nbessi for sure. The path has been long from 6.0 to 8.0 and lot of things remain to do. :sweat_smile: 
I think that once the odoo migration will be completed it will be time to upgrade OpenLayer in an other PR. This will provide a solid foundation for geo apps in V8.

lmignon committed:
[FIX] Fix issues when google is used as Raster Layer Type
* only render map when the anchor element is visible
* load google JS in web.assets_common

bwrsandman commented on base_geoengine/geo_db.py:
```diff
@@ -18,22 +18,36 @@
+        CREATE EXTENSION postgis_topology;
+    """)
+    except Exception, exc:
+        raise osv.except_osv(
```
These two lines are deprecated

It's recommended to use the `except Exception as exc` form with python2.6+
https://www.python.org/dev/peps/pep-3110/

`osv` is deprecated and `orm.except_orm` should be used (there are also v8 exceptions you can use, but I'm not too knowledgeable about them)

bwrsandman commented on base_geoengine/geo_db.py:
```diff
@@ -18,22 +18,36 @@
+        raise osv.except_osv(
+            _('Error, Can not initialize spatial postgis function.'
+              ' Database user may have to be superuser and postgres/postgis '
+              'extentions and dev header have to be installed'), str(exc))
```
Two things here, `_()` works by looking at the `context` variable in the local frame, if it's missing, there may not be a proper translation.

`str(exc)` might cause an `AsciiError` on non-english systems, use `ustr` from `openerp.tools` to avoid that.

For example: `str("Obélix")` causes an error in python2.

lmignon committed:
[FIX] Use an init hook to initialize postgis extension at install

lmignon committed:
[FIX] An other fix when google is used as Raster Layer Type
The call to animate on the div element before the map is rendered introduces an offset between the google map layer and others layers

Commit:
[IMP] Move icons to static/description.

lmignon committed:
Merge pull request #1 from foutoucour/8.0-migrate-from-7.0_move_icons

[IMP] Move icons to static/description.

lmignon committed:
[IMP] Extract description in __openerp__.py to README.rst

lmignon committed:
[FIX] Add group element to the view definition of retail_machine
The group element is required to properly display labels on the form

lmignon commented:
@bwrsandman The PR is ready to review.

lmignon committed:
[FIX] Get the tab id from the closest div with class='oe_notebook_page' element in place of the parent

bwrsandman commented on base_geoengine/README.rst:
```diff
@@ -0,0 +1,48 @@
+* Alexandre Fayolle <alexandre.fayolle@camptocamp.com>
+* Guewen Baconnier <guewen.baconnier@camptocamp.com>
+* Laurent Mignon <laurent.mignon@acsone.eu>
+* ...
```
The ellipses suggests that the list is incomplete

``` bash
$ git log --pretty="%an <%ae>"  base_geoengine | sort | uniq
Alexandre Fayolle @ camptocamp <alexandre.fayolle@camptocamp.com>
Frederic Junod <fredj@wrk5.wrk.lsn.camptocamp.com>
Guewen Baconnier @ Camptocamp <guewen.baconnier@camptocamp.com>
Laurent Mignon (aka lmi) <laurent.mignon@acsone.eu>
Laurent Mignon <laurent.mignon@acsone.eu>
nbessi and fredj <Unknown>
Nicolas Bessi <nbessi@gmail.com>
Nicolas Bessi <nicolas.bessi@camptocamp.com>
openerp <openerp@openerp-vagrant>
Sandy Carter <sandy.carter@savoirfairelinux.com>
Vincent Renaville <vincent.renaville@camptocamp.com>
Yannick Vaucher <yannick.vaucher@camptocamp.com>
```

nbessi replied:
nbessi is me nicolas.bessi@camptocamp.com and openerp@openerp-vagrant is aslo me.
Fredj is Frederic Junod frederic.junod@camptocamp.com

lmignon replied:
@nbessi it's fixed in  https://github.com/acsone/geospatial/commit/e3f05480740ecceba1b81f8d3923a6add912c63d

bwrsandman commented on base_geoengine/company.py:
```diff
@@ -19,20 +19,17 @@
 class ResCompany(geo_model.GeoModel):
     """Override company to show basic GeoEngine data"""
     _name = "res.company"
     _inherit = "res.company"
```
`_name` is redundant

bwrsandman commented on base_geoengine/company.py:
```diff
@@ -19,20 +19,17 @@
-ResCompany()
+    geoengine_base_info = fields.Text(
+        'General informations about GeoEngine', size=16, required=False,
+        readonly=True, default='NOT IMPLEMENTED YET')
```
What does this mean?
`Not implemented yet`

Could helptext be added?

lmignon replied:
This code is already there in 6.1 and not used... I'll remove it.

bwrsandman commented:
`geospatial/base_geoengine/data.xml` contains `res.groups` typically we put this in a `security/` directory

bwrsandman commented on base_geoengine/doc/source/conf.py:
```diff
@@ -11,23 +12,29 @@
+    os.path.abspath('/opt/openerp_61Launchpad/src/server/openerp/'))
+sys.path.insert(
+    0,
+    os.path.abspath('/opt/openerp_61Launchpad/src/server/openerp/addons/'))
```
Could this be abstracted. Hardcoded directories are not going to help anyone.

lmignon replied:
This should be fixed on the 6.1 and then forward ported into the others branches

bwrsandman replied:
OK, an issue should be created for all these, then.

bwrsandman commented on base_geoengine/fields.py:
```diff
@@ -0,0 +1,82 @@
+import logging
```
These definitely need docstrings.

The classes should explain what the types mean, for example.

lmignon replied:
done

bwrsandman commented on base_geoengine/geo_db.py:
```diff
@@ -18,22 +18,36 @@
-SaleOrder()
-
+def init_postgis(cr):
+    """ Initialize postgis
```
Could this be expanded to explain what is done and why?

For exmaple: create these extensions to enable this

bwrsandman commented on base_geoengine/geo_db.py:
```diff
@@ -18,22 +18,36 @@
+        raise MissingError(
+            'Error, Can not initialize spatial postgis function.'
+            ' Database user may have to be superuser and postgres/postgis '
+            'extentions and dev header have to be installed', ustr(exc))
```
Notpicking, but for consistence, add the spaces at the end of the lines. Also add a period at the end.

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -54,32 +56,31 @@ def set_geo(self, value):
 
-    def __init__(self, string, geo_type, dim=2, srid=900913 , gist_index=True, **args):
+    def __init__(self, string, dim=2, srid=900913, gist_index=True,
+                 **args):
```
Could you add a sphinx style docstring which explains the parameters and their function?

Then you can remove the comments for each assignment.

lmignon replied:
This should be fixed on the 6.1 and then forward ported into the others branches

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -54,32 +56,31 @@ def set_geo(self, value):
 
-
     def entry_to_shape(self, value, same_type=False):
         """Transform input into an object"""
```
How does it transform it? Why?

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -87,8 +88,8 @@ def entry_to_shape(self, value, same_type=False):
             if shape.geom_type.lower() != self._geo_type.lower():
                 msg = _('Geo Value %s must be of the same type %s as fields')
                 # XXX: Exception or TypeError ?
```
More precise error type is better than a less precice in my opinion.

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -103,26 +104,12 @@ def _create_index(self, cursor, table, col_name):
-                                 col_name, table)
+                logger.exception(
+                    'Cannot create gist index for col %s table %s:',
+                    col_name, table)
```
Is there a reason to not reraise or print out the error?

lmignon replied:
done

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -143,7 +130,8 @@ def create_geo_column(self, cursor, col_name, geo_column, table, model):
 
     def update_geo_column(self, cursor, col_name, geo_column, table, model):
-        """Update a column of type the geom does. !! not do a lot of test yet"""
+        """Update a column of type the geom does. !! not do a lot of test yet
```
This sentense needs to be reworked as it doesn't make much sense.

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -143,7 +130,8 @@ def create_geo_column(self, cursor, col_name, geo_column, table, model):
-        """Update a column of type the geom does. !! not do a lot of test yet"""
+        """Update a column of type the geom does. !! not do a lot of test yet
+        """
         query = ("SELECT srid, type, coord_dimension "
```
All the queries I've seen so far use triple quotes, this one is different, but it doesn't need to.

The advantage of triplestrings is that you avoid the issue of forgetting to append a space.

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -151,26 +139,34 @@ def update_geo_column(self, cursor, col_name, geo_column, table, model):
         check_data = cursor.fetchone()
         if not check_data:
-            raise Exception("geometry_columns table seems to be corrupted. SRID check is not possible")
+            raise Exception(
```
`TypeException`

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -227,7 +223,7 @@ def get(self, cr, obj, ids, name, uid=False, context=None, values=None):
 
 
-#We may use monkey patching but we do not want to break normal function field
+# We may use monkey patching but we do not want to break normal function field
```
Could this be in the docstring?

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -235,18 +231,21 @@ def postprocess(self, cr, uid, obj, field, value=None, context=None):
+
 class GeoFunction(fields.function):
-    #shell class
+    # shell class
```
Docstring?

Also could this be expanded?

bwrsandman commented on base_geoengine/geo_field.py:
```diff
@@ -235,18 +231,21 @@ def postprocess(self, cr, uid, obj, field, value=None, context=None):
+
 class GeoRelated(fields.related):
-    #shell class
+    # shell class
```
Docstring?

Also could this be expanded?

bwrsandman commented on base_geoengine/geo_helper/geo_convertion_helper.py:
```diff
@@ -34,12 +41,12 @@ def value_to_shape(value):
     elif hasattr(value, 'wkt'):
-        #Nasty but did not find equivalent of base string for shapely
+        # Nasty but did not find equivalent of base string for shapely
         if 'shapely.geometry' in str(type(value)):
```
`is_instance(value, shapely.geometry.base.BaseGeometry)`

bwrsandman commented on base_geoengine/geo_ir/ir_model.py:
```diff
@@ -38,7 +38,7 @@
 
 
-class IrModelField(osv.osv):
+class IrModelField(models.Model):
```
Model name is `ir.model.fields` so the filename should be `ir_model_fields.py` to be more consistant

bwrsandman commented on base_geoengine/geo_ir/ir_model.py:
```diff
@@ -48,16 +48,11 @@ def _get_fields_type(self, cr, uid, context=None):
+    geo_type = fields.Selection(
+        POSTGIS_GEO_TYPES, string="PostGIs type")
+    dim = fields.Selection(
+        [(2, '2'), (3, '3'), (4, '4')], string="PostGIs type")
```
Could a helptext be provided?

bwrsandman commented on base_geoengine/geo_ir/ir_model.py:
```diff
@@ -48,16 +48,11 @@ def _get_fields_type(self, cr, uid, context=None):
+        POSTGIS_GEO_TYPES, string="PostGIs type")
+    dim = fields.Selection(
+        [(2, '2'), (3, '3'), (4, '4')], string="PostGIs type")
+    gist_index = fields.Boolean('Create gist index')
```
Helptext for what cases would ask for indexing

bwrsandman commented on base_geoengine/geo_line.py:
```diff
@@ -18,17 +18,18 @@
-        res = super(GeoLine, self).__init__(string, "LINESTRING", dim=dim,
-                                            srid=srid, gist_index=gist_index, **args)
-        return res
+    def __init__(self, string, dim=2, srid=900913, gist_index=True, **args):
```
Docstring to explain the params would be nice. If it's from the super, make sure it is transfered.

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -18,35 +18,35 @@
 
-DEFAULT_EXTENT = '-123164.85222423, 5574694.9538936, 1578017.6490538, 6186191.1800898'
+DEFAULT_EXTENT = '-123164.85222423, 5574694.9538936, ' + \
+    '1578017.6490538, 6186191.1800898'
```
What do these values represent? It would be nice to have that in a comment.

Also, `+` is redundant

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -18,35 +18,35 @@
-    #Array of ash that define layer and data to use
+
+class GeoModel(models.BaseModel):
+    # Array of ash that define layer and data to use
```
This could be converted to a docstring.

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -18,35 +18,35 @@
-        ## We do this because actually creation of fields in DB is not actually
-        ## delegated to the field it self but to the ORM _auto_init function
+        # We do this because actually creation of fields in DB is not actually
+        # delegated to the field it self but to the ORM _auto_init function
```
Can these comments be integrated into the docstring?

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -18,35 +18,35 @@
-    _transient = False # True in a TransientModel
+    # not visible in ORM registry, meant to be python-inherited only
+    _register = False
+    _transient = False  # True in a TransientModel
```
Could the description of these attributes be added to the class docstring?

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -56,27 +56,21 @@ def _auto_init(self, cursor, context=None):
         return res
 
     def fields_get(self, cursor, uid, allfields=None, context=None):
         """Add geo_type definition for geo fields"""
```
Could the params be explained, especially `allfields`?

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -98,8 +92,9 @@ def _get_geo_view(self, cursor, uid):
         if not geo_view_id:
-            raise osv.except_osv(_('No GeoEngine view defined for the model %s') % (self._name,),
-                                 _('Please create a view or modify view mode'))
+            raise osv.except_osv(
```
`osv.except_osv` is deprecated, use a v8 exception

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -98,8 +92,9 @@ def _get_geo_view(self, cursor, uid):
-            raise osv.except_osv(_('No GeoEngine view defined for the model %s') % (self._name,),
-                                 _('Please create a view or modify view mode'))
+            raise osv.except_osv(
+                _('No GeoEngine view defined for the model %s') % self._name,
```
Will this work if `self._name = 'oblélix'` would that happen?

lmignon replied:
is 'obélix' allowed as model name?

bwrsandman replied:
If it happens, then the error message won't show, instead there will be an ASCII decode error

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -121,63 +117,72 @@ def set_field_real_name(in_tuple):
                 view = self._get_geo_view(cursor, uid)
             else:
                 view = view_obj.browse(cursor, uid, view_id)
```
Can you explain this condition? In what case `(not view_id) is True`?

lmignon replied:
comes from 6.1. I don't have investigated.

lmignon commented:
@bwrsandman Thank you for all your comments but are you aware that this PR is the result of the port from 6.1 to 8.0, with lot of things to fix in geospatial addons and also in odoo? Lots of your comments should be first addressed on the 6.1  and forward ported after on the others branches... Sure, there are a lot of possible improvements but these can be done in an other PR.

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -121,63 +117,72 @@ def set_field_real_name(in_tuple):
+        field = self._fields.get(column)
+        if field and not isinstance(field, geo_fields.GeoField):
+            raise ValueError(
+                _("%s column does not exists or is not a geo field") % column)
```
will this work if `column = 'astérix'` if that's a possible case, consider using `ustr`

lmignon replied:
is 'astérix' allowed as column name?

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -121,63 +117,72 @@ def set_field_real_name(in_tuple):
-        res['edit_raster'] = raster_obj.read(cursor, uid, raster_id[0], context=context)
-        res['geo_type'] = self._columns[column]._geo_type
-        res['srid'] = self._columns[column]._srid
+            raise osv.except_osv(
```
`osv.except_osv` is deprecated, use a v8 exception

bwrsandman commented on base_geoengine/geo_model.py:
```diff
@@ -121,63 +117,72 @@ def set_field_real_name(in_tuple):
-        res['srid'] = self._columns[column]._srid
+            raise osv.except_osv(
+                _('Configuration Error'),
+                _('No raster layer for view %s') % (view.name,))
```
use `ustr`

bwrsandman commented on base_geoengine/geo_multiline.py:
```diff
@@ -18,17 +18,18 @@
 ##############################################################################
-from __future__ import absolute_import
-from osv import fields
+from openerp.osv import fields
```
Can this be `from openerp import fields`?

bwrsandman commented on base_geoengine/geo_multipoint.py:
```diff
@@ -18,17 +18,18 @@
 ##############################################################################
-from __future__ import absolute_import
-from osv import fields
+from openerp.osv import fields
```
Can this be `from openerp import fields`?

lmignon replied:
no here your are in the column definition not in the field definittion

bwrsandman replied:
OK

bwrsandman commented on base_geoengine/geo_multiline.py:
```diff
@@ -18,17 +18,18 @@
-        res = super(GeoMultiLine, self).__init__(string, "MULTILINESTRING", dim=dim,
-                                                 srid=srid, gist_index=gist_index, **args)
-        return res
+    def __init__(self, string, dim=2, srid=900913, gist_index=True, **args):
```
Could you fetch the dosctring from the super?

bwrsandman commented on base_geoengine/geo_multipolygon.py:
```diff
@@ -18,16 +18,18 @@
 ##############################################################################
-from __future__ import absolute_import
-from osv import fields
+from openerp.osv import fields
```
Can this be `from openerp import fields`?

bwrsandman commented on base_geoengine/geo_multipolygon.py:
```diff
@@ -18,16 +18,18 @@
+    _geo_type = 'MULTIPOLYGON'
+
+    def __init__(self, string, dim=2, srid=900913, gist_index=True, **args):
+        super(GeoMultiPolygon, self).__init__(
```
Could you fetch the docstring from the super?

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -20,28 +20,36 @@
 
 logger = logging.getLogger('geoengine.sql.debug')
 
 # TODO Refactor geo_search and dry up the get_**_sql code
```
Is this done?

lmignon replied:
no

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -51,18 +59,21 @@ def geo_search(model, cursor, uid, domain=[], geo_domain=[], offset=0, limit=Non
      * geo_within
      * geo_intersect
     """
```
This is a very long function, could bits of it be factorized?

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
-    def _get_direct_como_op_sql(self, table, col, value, rel_col=None, rel_model=None, op=''):
-        "provide raw sql for geater and lesser operators"
+        except Exception:
+            raise Exception(
```
Could a more precice exception be caught?
Could a less generic exception be raised?

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
+
+    def _get_direct_como_op_sql(self, table, col, value, rel_col=None,
+                                rel_model=None, op=''):
+        """provide raw sql for geater and lesser operators"""
```
Could you provide sphinx param doc?

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
+            if rel_col and rel_model:
+                raise Exception(
+                    'Area %s does not support int compare for relation '
+                    'search' % (op,))
```
Use `ustr` on `op` and try raising a less vague exception

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
                 compare_to = base.wkt
-            return " ST_Area(%s.%s) %s ST_Area(ST_GeomFromText('%s'))" % (table, col, op, compare_to)
+            return " ST_Area(%s.%s) %s ST_Area(ST_GeomFromText('%s'))" % (
+                table, col, op, compare_to)
```
Use ustr where there might be ascii errors

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
-            compare_to = "ST_GeomFromText('%s')" %(base.wkt,)
-        return " %s(%s.%s, %s)" %(op, table, col, compare_to)
+            compare_to = "ST_GeomFromText('%s')" % (base.wkt,)
+        return " %s(%s.%s, %s)" % (op, table, col, compare_to)
```
Use ustr where there might be ascii errors

lmignon replied:
Ascii errors in this case? Do you think this can happen in this case?

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
 
-    ## Area comparison #############
-    def get_geo_greater_sql(self, table, col, value, rel_col=None, rel_model=None):
+    # Area comparison #############
```
Could this be put in the docstring?

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
                                             rel_col, rel_model, op='<')
 
-    ## Equality comparison #############
+    # Equality comparison #############
```
Could this be put in the docstring?

bwrsandman commented on base_geoengine/geo_operators.py:
```diff
@@ -132,69 +149,84 @@ def get_rel_field(self, rel_col, rel_model):
         return " %s.%s = %s" % (table, col, compare_to)
 
-    ## PostGis spatial comparison ###########
+    # PostGis spatial comparison ###########
```
Could this be put in the doctring?

bwrsandman commented on base_geoengine/geo_point.py:
```diff
@@ -18,16 +18,18 @@
-from __future__ import absolute_import
-from osv import fields
+
+from openerp.osv import fields
```
Can this be `from openerp import fields`?

bwrsandman commented on base_geoengine/geo_point.py:
```diff
@@ -18,16 +18,18 @@
+    _geo_type = 'POINT'
+
+    def __init__(self, string, dim=2, srid=900913, gist_index=True, **args):
+        super(GeoPoint, self).__init__(
```
Inheriting the docstring from super can help document what the parameters are supposed to be.

bwrsandman commented on base_geoengine/geo_polygon.py:
```diff
@@ -18,17 +18,18 @@
 ##############################################################################
-from __future__ import absolute_import
-from osv import fields
+from openerp.osv import fields
```
Can this be `from openerp import fields`?

bwrsandman commented on base_geoengine/geo_polygon.py:
```diff
@@ -18,17 +18,18 @@
-        res = super(GeoPolygon, self).__init__(string, "POLYGON",
-                                               dim=dim, srid=srid, gist_index=gist_index, **args)
-        return res
+    def __init__(self, string, dim=2, srid=900913, gist_index=True, **args):
```
Inheriting the docstring from super can help document what the parameters are supposed to be.

bwrsandman commented on base_geoengine/geo_view/geo_raster_layer.py:
```diff
@@ -0,0 +1,64 @@
+            'ir.ui.view', 'Related View', domain=[('type', '=', 'geoengine')],
+            required=True),
+        'use_to_edit': fields.boolean('Use to edit')}
+# TODO Write data check consraints
```
For what?

lmignon replied:
I'm not the author of this comment...

bwrsandman commented on base_geoengine/geo_view/geo_raster_layer.py:
```diff
@@ -0,0 +1,64 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import fields, orm
```
Shouldn't this be in v8 syntax?

bwrsandman commented:
@lmignon I'm just pointing out what I see.

If this is spam I can stop.

bwrsandman commented on base_geoengine_demo/README.rst:
```diff
@@ -0,0 +1,38 @@
+* Alexandre Fayolle <alexandre.fayolle@camptocamp.com>
+* Guewen Baconnier <guewen.baconnier@camptocamp.com>
+* Laurent Mignon <laurent.mignon@acsone.eu>
+* ...
```
```
$ git log --pretty="%an <%ae>"  base_geoengine_demo | sort | uniq
Guewen Baconnier @ Camptocamp <guewen.baconnier@camptocamp.com>
Jordi Riera <jordi.riera@savoirfairelinux.com>
Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>
Laurent Mignon (aka lmi) <laurent.mignon@acsone.eu>
Laurent Mignon <laurent.mignon@acsone.eu>
nbessi and fredj <Unknown>
Nicolas Bessi <nbessi@gmail.com>
Nicolas Bessi <nicolas.bessi@camptocamp.com>
Sandy Carter <sandy.carter@savoirfairelinux.com>
Yannick Vaucher <yannick.vaucher@camptocamp.com>
```

bwrsandman commented on base_geoengine_demo/zip_view.xml:
```diff
@@ -81,7 +81,7 @@
             <field name="begin_color">#FF680A</field>
             <field name="end_color">#00680A</field>
-        </record>
+        </record >
```
This probably doesn't need to be commited :wink:

bwrsandman commented on geoengine_geoname_geocoder/README.rst:
```diff
@@ -0,0 +1,53 @@
+* Alexandre Fayolle <alexandre.fayolle@camptocamp.com>
+* Guewen Baconnier <guewen.baconnier@camptocamp.com>
+* Laurent Mignon <laurent.mignon@acsone.eu>
+* ...
```
```
$ git log --pretty="%an <%ae>"  geoengine_geoname_geocoder | sort | uniq
Alexandre Fayolle @ camptocamp <alexandre.fayolle@camptocamp.com>
Guewen Baconnier @ Camptocamp <guewen.baconnier@camptocamp.com>
Jordi Riera <jordi.riera@savoirfairelinux.com>
Laurent Mignon (ACSONE) <laurent.mignon@acsone.eu>
Laurent Mignon (aka lmi) <laurent.mignon@acsone.eu>
Laurent Mignon <laurent.mignon@acsone.eu>
Nicolas Bessi <nbessi@gmail.com>
Nicolas Bessi <nicolas.bessi@camptocamp.com>
Sandy Carter <sandy.carter@savoirfairelinux.com>
```

bwrsandman commented on geoengine_geoname_geocoder/__openerp__.py:
```diff
@@ -18,19 +18,22 @@
  'active': False,
- 'icon': '/base_geoengine/static/src/images/map_icon.png'}
+ }
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
You can remove the vim line too

bwrsandman commented on geoengine_geoname_geocoder/res_partner.py:
```diff
@@ -94,31 +106,28 @@ def geocode_from_geonames(self, cursor, uid, ids, srid='900913', strict=True, co
+        if do_geocode \
+            and "country_id" in vals \
+            or 'city' in vals \
+                or 'zip' in vals:
```
This isn't pep8

try

``` python
        if do_geocode and set(['country', 'city', 'zip']).issubset(vals):
```

or

``` python
        if do_geocode and {'country', 'city', 'zip'}.issubset(vals):
```

lmignon replied:
Why it's not pep8? Travis is happy with this syntax...

bwrsandman replied:
About logical operators needing to be at the end of a line.

Maybe the `\` makes it so pep8 doesn't register it.

Anyway, isn't it simpler to use `issubset`?

bwrsandman commented on geoengine_geoname_geocoder/wizard/bulk_encode.py:
```diff
@@ -19,22 +19,24 @@
 ##############################################################################
 
-from osv import fields, osv
+from openerp.osv import fields, orm
```
Can v8 api be used?

bwrsandman commented on geoengine_partner/README.rst:
```diff
@@ -0,0 +1,39 @@
+* Alexandre Fayolle <alexandre.fayolle@camptocamp.com>
+* Guewen Baconnier <guewen.baconnier@camptocamp.com>
+* Laurent Mignon <laurent.mignon@acsone.eu>
+* ...
```
```
$ git log --pretty="%an <%ae>"  geoengine_partner | sort | uniq
Guewen Baconnier @ Camptocamp <guewen.baconnier@camptocamp.com>
Jordi Riera <jordi.riera@savoirfairelinux.com>
Laurent Mignon (aka lmi) <laurent.mignon@acsone.eu>
Laurent Mignon <laurent.mignon@acsone.eu>
Nicolas Bessi <nbessi@gmail.com>
Nicolas Bessi <nicolas.bessi@camptocamp.com>
Sandy Carter <sandy.carter@savoirfairelinux.com>
```

bwrsandman commented on geoengine_sale/README.rst:
```diff
@@ -0,0 +1,39 @@
+* Alexandre Fayolle <alexandre.fayolle@camptocamp.com>
+* Guewen Baconnier <guewen.baconnier@camptocamp.com>
+* Laurent Mignon <laurent.mignon@acsone.eu>
+* ...
```
```
$ git log --pretty="%an <%ae>"  geoengine_sale | sort | uniq
Guewen Baconnier @ Camptocamp <guewen.baconnier@camptocamp.com>
Jordi Riera <jordi.riera@savoirfairelinux.com>
Laurent Mignon (aka lmi) <laurent.mignon@acsone.eu>
Laurent Mignon <laurent.mignon@acsone.eu>
Nicolas Bessi <nbessi@gmail.com>
Nicolas Bessi <nicolas.bessi@camptocamp.com>
Sandy Carter <sandy.carter@savoirfairelinux.com>
```

lmignon committed:
[FIX] remove unused code

bwrsandman commented:
Sorry for the spam, I went through the entire diff pointing out what could be improved.

lmignon committed:
[FIX] move the group definitions into the security folder

lmignon committed:
[FIX] list all know contributors from git

lmignon commented:
@bwrsandman can you change the labels?

lmignon committed:
[IMP]Add documentation for new fields

lmignon committed:
[IMP]Add documentation for the init_hook

lmignon committed:
[FIX] Reraise error

lmignon committed:
[FIX] Use a more precise type error

lmignon committed:
[IMP] Use triple quotes for sql query

lmignon committed:
[FIX] Reworks docstring

lmignon committed:
[FIX] Use a more precise type error

lmignon committed:
[FIX] + is redundant

lmignon committed:
[IMP] Add documentation

lmignon committed:
[FIX] Use a more precise exception

lmignon committed:
[FIX] Reworks docstring

lmignon committed:
[IMP] Use V8 API

lmignon committed:
[FIX] Removes the remaining vim line

lmignon committed:
[FIX] Typo

lmignon committed:
[IMP] Use V8 API

lmignon commented:
@bwrsandman I've addressed lot of your comments. I hopes that the remaining issues are not blocking. If these are, any help will be appreciated to fix them. I've spent lot of time to migrate these addons and I've no more time to fix the issues already there in the 6.1 branch.

lmignon committed:
[ADD] New module extending base_geolocalize
The modul add a new field 'geo_point' computed from partner_latitue and partner_longitude added by base_geolocalize. The geo_point is rendered in a map on the partner form

lmignon committed:
[IMP] Pep8

lmignon commented:
@bwrsandman is the 'needs fixing' label still relevant?

bwrsandman commented:
@lmignon
Things that still worry me:
- [x] Doc that could be improved
- [x] Code which whose function we're unsure of 
- [x] Exceptions which use `%s` formating without a `ustr` cast that might raise `AsciiErrors` instead
- [x] `is_instance(value, shapely.geometry.base.BaseGeometry)` to replace string compare
- [x] Naming of some files which don't match the model
- [x] Failing tests ("External ID not found in the system: base_geoengine_demo.field_dummy_zip_total_sales")

We can move some of these fixes to other issues so they can be done across all branches.
Let me know which ones, we'll create dedicated issues.

lmignon commented:
@bwrsandman 
- Doc is the same as in 6.1. It can be improved in an other PR.

> [ ] Code whose function we're unsure of 

As a contributor, can you explain what you don't understand? `entry_to_shape(self, value, same_type=False):` is used to transform a string representing a geometry object into python Shapely object. As result of this transformation each time you access to a geo field on the server side, the value is always a Geometry object. (never a string). @nbessi am I right?

> [ ]Exceptions which use %s formating without a ustr cast that might raise AsciiErrors instead

The formatted values into the  message of the referenced exceptions are all related to column, table or operator... Do you have tried to use non ascii character for one of these values? 

> [ ] is_instance(value, shapely.geometry.base.BaseGeometry) to replace string compare

OK

> [ ] Naming of some files which don't match the model

The naming is consistent across branches. Moreover the base model is defined in the file openerp/addons/base/ir/ir.model.py 

> [ ] Failing tests ("External ID not found in the system: base_geoengine_demo.field_dummy_zip_total_sales")

As explained in the PR description, two bugs in odoo prevent a proper use of the module. The reported error is related to https://github.com/odoo/odoo/issues/5616 (PR odoo/odoo#5618).

lmignon commented:
sorry I've pressed the wrong button when submitting my last comment

bwrsandman commented:
No worries. Thank you for the explanation.

bwrsandman commented:
For `entry_to_shape`, it is actually a question for more doc.

The code we're unsure of is this:
https://github.com/OCA/geospatial/pull/24#discussion_r26137063

lmignon committed:
[FIX] Use isinstance operator in place of the previous hack
Shapely now provide a BaseGeometry class as base class for all Geometry objects

lmignon commented:
@bwrsandman see https://github.com/OCA/geospatial/commit/d603fa7e7e8cea895e2aa66e3077c66b16bfdcd2 
@nbessi should be able to say if the code is still required.

bwrsandman commented:
@lmignon Not a whole lot of explanation in that commit. It'll be another documentation point to add.

bwrsandman commented:
:+1: Code review

lmignon committed:
[FIX] add missing api decorator

Commit:
[FIX] run tests in V8

Commit:
[FIX] With new API: Non stored fields doesn't have column attribute

foutoucour commented:
:+1: 
Javascripts would need more doc tho

bwrsandman commented:
Is this ready to be merged? I see enough :+1:

lmignon commented:
@bwrsandman Yes it is IMO. But if it's merged now, we must change the README to warn about the 2 bugs in Odoo preventing a proper use.

nbessi commented on base_geoengine/geo_model.py:
```diff
@@ -18,35 +18,42 @@
         columns = {}
         geo_columns = {}
         tmp = {}
-        #geo_db.init_postgis(cursor)
```
dead comment

lmignon replied:
@nbessi postgis extension is installed by a 'pre_init_hook' at module install. The commented line is a remainder of a previous change and is now removed.

bwrsandman replied:
If that's the case, your comment of "remainder of a previous change and is now removed" should be in the code to provide the context

lmignon replied:
@bwrsandman sorry,  I don't understand your comment. The commented line is removed.

bwrsandman replied:
If someone reads this, they will surely come to the same conclusion as @nbessi 

A simple comment in the code explaining, like you did in your comment, why it is still there is appropriate to make the code easier to maintain and understand.

lmignon replied:
@bwrsandman on which line do you want a comment?? 
`geo_db.init_postgis(cursor)` is removed from the code. The line no more exists.

nbessi commented on geoengine_geoname_geocoder/README.rst:
```diff
@@ -0,0 +1,54 @@
+* Jordi Riera <jordi.riera@savoirfairelinux.com>
+* Laurent Mignon <laurent.mignon@acsone.eu>
+* Nicolas Bessi <nicolas.bessi@camptocamp.com>
+* Sandy Carter <sandy.carter@savoirfairelinux.com>
```
It is missing some GIS Camptocamp developer here that wrote the first JavaScript support.
The original authors ar:

Frédéric Junod
Bruno Binet
Nicolas Bessi

And Also 
Yannick Vaucher is missing as a contributor

I guess it should be OK as there are under the Camptocamp CLA.

yvaucher replied:
I think I contributed to `base_geoengine` but not to `geoengine_geoname_geocoder`

lmignon committed:
[FIX] Add missing authors

lmignon committed:
[ADD] Add acsone in the list of authors

nbessi commented:
I should have some time planned to make an effort on this PR in the next two weeks in term of function validation and improvement. 

Before merging this PR we should update the documentation that is quite outdated. 
Especially the PostGIS part that is totally obsolete. 

We should also move the doc folder to the root of the project and add a build script in Travis to test that sphinx doc is correct and will build. 

I'll try to work on this.

lmignon commented:
@nbessi Thank you for the review. I'll be present to the OCA Sprint before Odoo Experience 2015.  That could be a good place to work on the remaining issues and to finalise the migration to Odoo 8.

nbessi commented:
@lmignon yes it should be a nice opportunity to give a boost to the addons suite

nbessi committed:
Fix MissingError wrong argument number

And improve error message

nbessi committed:
Improve error message

nbessi committed:
Styling

nbessi commented:
@lmignon I have started to work on your work. We have a nasty situation here. The initialisation of the colums is made with old api model, but fields can be of new api type etc. I propose to transfer all the know how to the new api fields and to base all the initialization of fields base on the _fields property.

I also propose to implement a dedicated _field_create for geofields that shoud be clean and fast.

I also really hesitate to keep old style field. At least it should only contains read write functionality but initialization should always been delegated to new style fields, it should not be to complicated.

Also the new style fields do not seem to be called. I have to find why...

lmignon commented:
@nbessi The new api define a clean separation between field definition and columns. For example, non-stored fields doesn't have a column instance if they are declared with the new API. IMO the initialization of the column must stay based on the column definition.
 We have models with geo_field declared with the new and the old API and everything works fine. Do you have applied the 2 PRs on your Odoo instance (see PR description)?

nbessi commented:
@lmignon Yes I saw the patches. I'm afraid  it is not a concern separation but just the result of an intermediate situation due to API compatibility... If your interpretation is correct I agree with you, we should keep the code here.

I'm also looking to see if  can not do the st_geometry conversion in Python instead using PostGIS. It can be a nice workaround to odoo/odoo#5620 and we may gain some performance.

I'm also testing the browse behavior with self.pool and self.env and old style, new style model to see if we have a coherent model. PR should arrive soon.

nbessi commented:
@lmignon I think I will set the sales_total function field to store as long your PR is not merged.
It should at least enable demo module to works out of the box. Do you agree?

lmignon commented:
@nbessi I agree if we also reference this limitation in the description.

nbessi commented:
@lmignon I have found some work issue I'll try to fix on this pr:
- search on map view raise an exception
- pan does not work if screen full of vector
- selection in map view is  not persistant when switching back to list view.

lmignon commented:
@nbessi  thank you to keep me in touch!

nbessi committed:
Fix styling

nbessi commented:
other issue found:
CSV export/import is odd (fixed)
On change have to be tested

nbessi committed:
Style

nbessi committed:
Experiment to remove of history

nbessi committed:
modify computed field to be a stored field

This is a workaround of issue https://github.com/odoo/odoo/pull/5618
Where non stored fields are not set into ir.models.fields table

nbessi committed:
Add type testing when browsing

nbessi committed:
Revert function field is stored.

As long wa do not have implemented a geodepends decorator
Store values will never be refreshed

nbessi committed:
Avoid exception when query result is empty on map view

nbessi committed:
White space cleanup

nbessi committed:
Add search view on dummy zip

nbessi committed:
Remove experimental dead code

nbessi committed:
Remove obsolete view attributes

nbessi committed:
Add search view on retail

nbessi committed:
Fix select to index

nbessi commented:
@lmignon I fixed most of the issues here:

https://github.com/acsone/geospatial/pulls

It should give us a good starting point for the code sprint

lmignon commented:
@nbessi Than you, PR merged.

lmignon committed:
[IMP] Merge required patches on Odoo

lmignon committed:
[FIX] avoid checking SSL certificate on wget

lmignon committed:
[FIX] Use env var to build the path to the Odoo checkout

lmignon committed:
Only apply patch when building Odoo

lmignon committed:
[FIX] patch the right directory

lmignon committed:
[FIX] Sources for odoo are located in /home/lmi/odoo

lmignon committed:
[FIX] try to get the right repo name

lmignon committed:
[FIX] Use absolute path to patch file

lmignon committed:
[FIX] typo

lmignon committed:
[FIX] force patch command

lmignon committed:
[IMP]Update travis configuration
Follow OCA/maintainer-quality-tools#187 and use addons/apt/package directive to
install our build dependencies. This allows using container based Travis builds
(which should be faster).

lmignon committed:
[FIX] tests

yvaucher commented:
:fireworks:
